public/components/toolbar/spatial_filter/display_draw_helper.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiButton, EuiFlexGroup, EuiFlexItem, EuiPanel } from '@elastic/eui';
+import React, { memo, useEffect } from 'react';
+import { i18n } from '@osd/i18n';
+import { Map as Maplibre } from 'maplibre-gl';
+import { FILTER_DRAW_MODE } from '../../../../common';
+
+interface DrawFilterShapeHelper {
+  map: Maplibre;
+  mode: FILTER_DRAW_MODE;
+  onCancel: () => void;
+}
+
+const getHelpText = (mode: FILTER_DRAW_MODE): string => {
```

Comment:
Can we use useMemo to avoiding unnecessary re-renders since

Reply:
sure

Reply:
I already added Memo to component. This method is outside component, it won't be generated everytime.